File: rb/features.rb
# Tvmaze SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TvmazeFeatures
  def self.make_feature(name)
    case name
    when "base"
      TvmazeBaseFeature.new
    when "test"
      TvmazeTestFeature.new
    else
      TvmazeBaseFeature.new
    end
  end
end

File: py/features.py
# Tvmaze SDK feature factory

from feature.base_feature import TvmazeBaseFeature
from feature.test_feature import TvmazeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TvmazeBaseFeature(),
        "test": lambda: TvmazeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
